app/src/main/java/org/mozilla/fenix/ext/BrowserIcons.kt

```diff
@@ -4,8 +4,35 @@
 
 package org.mozilla.fenix.ext
 
+import android.graphics.drawable.Drawable
+import android.view.View
 import android.widget.ImageView
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import mozilla.components.browser.icons.BrowserIcons
 import mozilla.components.browser.icons.IconRequest
 
-fun BrowserIcons.loadIntoView(view: ImageView, url: String) = loadIntoView(view, IconRequest(url))
+fun BrowserIcons.loadIntoView(
+    view: ImageView,
+    url: String = "",
+    localImage: Drawable? = null,
+    iconCover: ImageView? = null
+): Job? {
+    if (localImage != null) view.setImageDrawable(localImage)
+    GlobalScope.launch {
```

Comment:
With `GlobalScope` this can live longer than the activity/fragment and keeps a reference to views, so potentially leaks?

CC @csadilek